File: cs315_codes/ch10_examples/C/nested-blocks.c
#include<stdio.h>
main() {
  int i=1;
  printf("In main i=%d\n", i);
  { // outer block
    int i=2;
    printf("In outer block i=%d\n", i);
    { // inner block
      int i=3;
      printf("In inner block i=%d\n", i);
    }
    printf("In outer block i=%d\n", i);
  }
  printf("In main i=%d\n", i);
}

File: cs315_codes/ch10_examples/Java/NestedBlocks.java
public class NestedBlocks {
	public static void main(String[] args) {
		int i=1;
		System.out.println("In main i="+i);
		{ // outer block
			int i=2;
			System.out.println("In outer block i="+i);
			{ // inner block
				int i=3;
				System.out.println("In outer block i="+i);
			}
			System.out.println("In outer block i="+i);
		}
		System.out.println("In main i="+i);
	}
}
